runner/internal/metrics/type.go

```diff
@@ -0,0 +1,49 @@
+package metrics
+
+import (
+	"reflect"
+	"time"
+)
+
+// Type represents the underlying type of a Value.
+type Type uint8
+
+const (
+	lastGoReflectKind = reflect.UnsafePointer
+
+	// TypeInvalid corresponds to an invalid type.
+	TypeInvalid = Type(reflect.Invalid)
+	// TypeInt corresponds to type int.
+	TypeInt = Type(reflect.Int)
+	// TypeDuration corresponds to type time.Duration.
+	TypeDuration = Type(lastGoReflectKind + iota)
```

Comment:
I'm just not really sure what you did here with `lastGoReflectKind`?

Reply:
reflect kinds are numbers (`uint`), and I wanted to match Go's values for `metrics.Type`. However `TypeDuration` has no equivalent in `reflect`, so I took the last Go `reflect` kind + 1 .
Anyway all of this is gone now!